execution_engine_testing/tests/bin/disk_use.rs

```diff
@@ -71,115 +93,148 @@ pub fn multiple_native_transfers(
+    engine_state::Error: From<S::Error>,
+{
+    let mut queue = Vec::new();
+    queue.push(state_root);
+
+    while let Some(root) = queue.pop() {
+        if necessary_tries.contains(&root) {
+            continue;
+        }
+        necessary_tries.insert(root);
 
-    let mut byte_size_map = HashMap::new();
-    let mut time_ms_map = HashMap::new();
+        let trie_or_chunk: TrieOrChunk = engine_state
+            .get_trie(CorrelationId::new(), TrieOrChunkId(0, root))
+            .unwrap()
+            .expect("trie should exist");
 
-    for (column, values) in results.iter() {
-        let bytes_values: Vec<usize> = values
-            .iter()
-            .map(|(_, byte_size, _, _)| *byte_size)
-            .collect();
-        byte_size_map.insert(column.clone(), bytes_values);
-        let time_ms_values: Vec<usize> = values.iter().map(|(_, _, _, time_ms)| *time_ms).collect();
-        time_ms_map.insert(column.clone(), time_ms_values);
-    }
+        let trie_bytes = match trie_or_chunk {
+            TrieOrChunk::Trie(trie) => trie,
+            TrieOrChunk::ChunkWithProof(_) => continue,
+        };
 
-    // bytes
-    println!("bytes report");
-    for (column, _) in results.iter() {
-        write!(&mut bytes_report, "{}-bytes,", column).unwrap();
-    }
-    writeln!(&mut bytes_report, "transfers").unwrap();
-
-    for (idx, transfer_count) in transfer_counts.iter().enumerate() {
-        for (column, _) in results.iter() {
-            write!(&mut bytes_report, " {},", byte_size_map[column][idx]).unwrap();
+        if let Some(0) = trie_bytes.get(0) {
+            continue;
         }
-        writeln!(&mut bytes_report, " {}", transfer_count).unwrap();
-    }
 
-    println!("time_ms report");
-    //
-    // Time ms
-    for (column, _) in results.iter() {
-        write!(&mut time_report, "{}-time_ms,", column).unwrap();
-    }
-    writeln!(&mut time_report, "transfers").unwrap();
-    for (idx, transfer_count) in transfer_counts.iter().enumerate() {
-        for (column, _) in results.iter() {
-            write!(&mut time_report, " {},", time_ms_map[column][idx]).unwrap();
+        let trie: Trie<Key, StoredValue> =
+            bytesrepr::deserialize(trie_bytes.inner_bytes().to_owned())
+                .expect("unable to deserialize");
+
+        match trie {
+            Trie::Leaf { .. } => continue,
+            Trie::Node { pointer_block } => queue.extend(pointer_block.as_indexed_pointers().map(
+                |(_idx, ptr)| match ptr {
+                    Pointer::LeafPointer(digest) | Pointer::NodePointer(digest) => digest,
+                },
+            )),
+            Trie::Extension { affix: _, pointer } => match pointer {
+                Pointer::LeafPointer(digest) | Pointer::NodePointer(digest) => queue.push(digest),
+            },
         }
-        writeln!(&mut time_report, " {}", transfer_count).unwrap();
     }
+}
 
-    println!("wrote reports");
+fn main() {
```

Comment:
I suggest we make this a `#[test]` so it gets run in CI (on a small scale).

Perhaps we could make it configurable with an environment parameter so we could run a longer nightly test to check for regressions?

Or maybe we just do the latter...

Reply:
I did this in https://github.com/casper-network/casper-node/pull/2877